### PR TITLE
Add Round5

### PR DESCRIPTION
**Pull request by TomerHawk**
Not sure if still in interest because Round5 didn't reach NIST round3. 

This version is without description of the two flavours. Because I am not sure which kind of content should fill the description field.

### CODE REVIEW

File: encryption/round5/round5-kem/bench/m4_r5nd-5kem-0d_m4.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+platform: Cortex M4
```
Comment (dingens) [inline encryption/round5/round5-kem/bench/m4_r5nd-5kem-0d_m4.yaml:1]: lacks values

File: encryption/round5/round5-kem/bench/avx2_r5n1-3kem-0d_opt.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+sources:
+  - ''
+platform: 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (dingens) [inline encryption/round5/round5-kem/bench/avx2_r5n1-3kem-0d_opt.yaml:3]: ehr :)

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+platform: 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (dingens) [inline encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml:1]: üöh

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+sources:
+  - ''
+platform: 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (dingens) [inline encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml:3]: hng

**Review by dingens — DISMISSED** [review]
Ah, I forgot. This PR removes stuff in `encryption/lac`. I don't know what you did there but that's wrong :)

**Comment by TomerHawk** [comment]
> Ah, I forgot. This PR removes stuff in `encryption/lac`. I don't know what you did there but that's wrong :)

Oh I will check this. 

> I've fixed a few numbers. Can you review my commit?
> A few platform statements are missing (confused with the source entries), can you fix that?

Sure :)

File: encryption/round5/round5-kem/bench/avx2_r5n1-3kem-0d_opt.yaml (hunk 2 above)
Comment (TomerHawk) [reply encryption/round5/round5-kem/bench/avx2_r5n1-3kem-0d_opt.yaml]: ups

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml (hunk 3 above)
Comment (TomerHawk) [reply encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml:1]: hui

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml (hunk 4 above)
Comment (TomerHawk) [reply encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml:3]: mmh

File: encryption/round5/round5-kem/bench/m4_r5nd-5kem-0d_m4.yaml (hunk 1 above)
Comment (TomerHawk) [reply encryption/round5/round5-kem/bench/m4_r5nd-5kem-0d_m4.yaml:1]: Oh I missed that, thanks

**Comment by TomerHawk** [comment]
> Very nice!
> 
> I've fixed a few numbers. Can you review my commit?
> 
> A few platform statements are missing (confused with the source entries), can you fix that?
> 
> I've skipped checking the benchmark numbers, because they're soooo many ;) and I don't think it's that important as it's not in round 3 anyways...

Thank you for your review. I reviewed your commits and they are on point :+1:  I probably got tired eyes or something :)

**Comment by dingens** [comment]
Can you add values to the `platform` fields that you just entered? It should be required, actually, I don't know why it is.

Apart from that we can merge! (Someone else please do it or I'll do it the week after next :) )

File: encryption/round5/round5-pke/bench/avx2_r5n1-1pke-0d_opt.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-1pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+platform: ''
+timings:
+  gen: 223000000
+  enc|sign: 1729000
+  dec|vrfy: 7338000
+  unit: cycles
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml:8]: `#`

File: encryption/round5/round5-pke/bench/avx2_r5n1-5pke-0d_opt.yaml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-5pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/m4_r5n1-3pke-0smallct_m4.yaml [hunk 9]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/m4_r5n1-3pke-0smallct_m4.yaml:1]: missing some data

File: encryption/round5/round5-pke/bench/opt_r5n1-1pke-0d_opt.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 62'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5n1-1pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5n1-3pke-0d_opt.yaml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 62'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5n1-3pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5n1-3pke-0smallct_opt.yaml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 63'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5n1-3pke-0smallct_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5n1-5pke-0d_opt.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 62'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5n1-5pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/param/r5nd-5pke-5d.yaml [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 5
+failure probability: -227
+sizes:
+  sk: 1042
+  pk: 978
+  ct|sig: 1301
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5nd-5pke-5d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5nd-5pke-0d.yaml [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 3
+failure probability: -143
+sizes:
+  sk: 1413
+  pk: 1349
+  ct|sig: 1525
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5nd-5pke-0d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5nd-3pke-5d.yaml [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 3
+failure probability: -256
+sizes:
+  sk: 828
+  pk: 780
+  ct|sig: 950
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5nd-3pke-5d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5nd-3pke-0d.yaml [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 3
+failure probability: -147
+sizes:
+  sk: 1031
+  pk: 983
+  ct|sig: 1119
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5nd-3pke-0d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5nd-1pke-5d.yaml [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 1
+failure probability: -142
+sizes:
+  sk: 493
+  pk: 461
+  ct|sig: 636
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5nd-1pke-5d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5nd-1pke-0d.yaml [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 1
+failure probability: -155
+sizes:
+  sk: 708
+  pk: 676
+  ct|sig: 756
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5nd-1pke-0d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5n1-5pke-0d.yaml [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 5
+failure probability: -144
+sizes:
+  sk: 14700
+  pk: 14636
+  ct|sig: 14724
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5n1-5pke-0d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5n1-3pke-0smallct.yaml [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 3
+failure probability: -149
+sizes:
+  sk: 163584
+  pk: 163536
+  ct|sig: 988
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5n1-3pke-0smallct.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5n1-3pke-0d.yaml [hunk 22]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 3
+failure probability: -144
+sizes:
+  sk: 9708
+  pk: 9660
+  ct|sig: 9732
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5n1-3pke-0d.yaml:14]: `#`

File: encryption/round5/round5-pke/param/r5n1-1pke-0d.yaml [hunk 23]
```diff
@@ -0,0 +1,15 @@
...
+  nist category: 1
+failure probability: -146
+sizes:
+  sk: 5772
+  pk: 5740
+  ct|sig: 5804
+number of operations: inf
+sources:
+  - >-
+    https://round5.org/Supporting_Documentation/Round5_Submission.pdf , page
```
Comment (taudor) [inline encryption/round5/round5-pke/param/r5n1-1pke-0d.yaml:14]: `#`

File: encryption/round5/round5-pke/impl/m4.yaml [hunk 24]
```diff
@@ -0,0 +1,6 @@
+name: M4
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 67'
```
Comment (taudor) [inline encryption/round5/round5-pke/impl/m4.yaml:3]: `#`

File: encryption/round5/round5-pke/impl/avx2.yaml [hunk 25]
```diff
@@ -0,0 +1,7 @@
+name: AVX2
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 64'
```
Comment (taudor) [inline encryption/round5/round5-pke/impl/avx2.yaml:3]: `#`

File: encryption/round5/round5-pke/bench/opt_r5nd-5pke-5d_opt.yaml [hunk 26]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 61'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5nd-5pke-5d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5nd-5pke-0d_opt.yaml [hunk 27]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 60'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5nd-5pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5nd-3pke-5d_opt.yaml [hunk 28]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 61'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5nd-3pke-5d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5nd-3pke-0d_opt.yaml [hunk 29]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 60'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5nd-3pke-0d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5nd-1pke-5d_opt.yaml [hunk 30]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 61'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5nd-1pke-5d_opt.yaml:2]: `#`

File: encryption/round5/round5-pke/bench/opt_r5nd-1pke-0d_opt.yaml [hunk 31]
```diff
@@ -0,0 +1,15 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf, page 60'
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/opt_r5nd-1pke-0d_opt.yaml:2]: `#`

File: README.md [hunk 32]
```diff
@@ -14,8 +14,10 @@ A comprehensive list of post-quantum crypto schemes and their properties.
 * [NewHope](encryption/newhope) ([Website](https://www.newhopecrypto.org/))
 * [NTRU Prime](encryption/ntru-prime) ([Website](https://ntruprime.cr.yp.to/))
 * [NTRU](encryption/ntru) ([Website](https://ntru.org/))
+* [Round5](encryption/round5) ([Website])(https://round5.org/)
```
[the remark this answers is not in the document]
Comment (TomerHawk) [reply README.md:17]: I am on it

File: encryption/round5/round5-pke/bench/m4_r5n1-3pke-0smallct_m4.yaml (hunk 9 above)
Comment (TomerHawk) [reply encryption/round5/round5-pke/bench/m4_r5n1-3pke-0smallct_m4.yaml:1]: There aren't vaules for this benchmark. I will remove it

File: encryption/round5/round5-kem/bench/avx2_r5n1-1kem-0d_opt.yaml [hunk 33]
```diff
@@ -0,0 +1,7 @@
+sources:
```
Comment (taudor) [inline encryption/round5/round5-kem/bench/avx2_r5n1-1kem-0d_opt.yaml:1]: @TomerHawk do you think we can conclude from subsection 2.9.9 that the same platform was used to run the avx2 benchmarks like for the optimized implementation? Hardware would support it, I don't see why they would change it.

File: encryption/round5/round5-kem/bench/avx2_r5n1-3kem-0d_opt.yaml [hunk 34]
```diff
@@ -0,0 +1,8 @@
+platform: ''
```
Comment (taudor) [inline encryption/round5/round5-kem/bench/avx2_r5n1-3kem-0d_opt.yaml:1]: see other comment

File: encryption/round5/round5-kem/bench/avx2_r5n1-5kem-0d_opt.yaml [hunk 35]
```diff
@@ -0,0 +1,7 @@
+sources:
```
Comment (taudor) [inline encryption/round5/round5-kem/bench/avx2_r5n1-5kem-0d_opt.yaml:1]: see other comment

File: encryption/round5/round5-pke/bench/avx2_r5n1-1pke-0d_opt.yaml [hunk 36]
```diff
@@ -0,0 +1,7 @@
+sources:
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-1pke-0d_opt.yaml:1]: see other comment (missing `platform`)

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml [hunk 37]
```diff
@@ -0,0 +1,8 @@
+sources:
+  - 'https://round5.org/Supporting_Documentation/Round5_Submission.pdf#table.20'
+platform: ''
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0d_opt.yaml:3]: see other comment (missing `platform`)

File: encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml [hunk 38]
```diff
@@ -0,0 +1,8 @@
+platform: ''
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-3pke-0smallct_opt.yaml:1]: see other comment (missing `platform`)

File: encryption/round5/round5-pke/bench/avx2_r5n1-5pke-0d_opt.yaml [hunk 39]
```diff
@@ -0,0 +1,7 @@
+sources:
```
Comment (taudor) [inline encryption/round5/round5-pke/bench/avx2_r5n1-5pke-0d_opt.yaml:1]: see other comment (missing `platform`)

File: encryption/round5/round5-kem/bench/avx2_r5n1-1kem-0d_opt.yaml (hunk 33 above)
Comment (TomerHawk) [reply encryption/round5/round5-kem/bench/avx2_r5n1-1kem-0d_opt.yaml:1]: You are right : "The performance was measured the same  way  as  the  non-AVX2  optimized  implementation "

**Comment by dingens** [comment]
I have checked the overall structure and everything, but among the benchmarks only the `avx2` ones, not the opt and m4 because I deemed the former more important and because it's so much data and not in round 3 anymore. If tudor is fine as well and you all are fine with my last two commits, we can merge.

**Review by taudor — APPROVED** [review]

**Pull request merged**